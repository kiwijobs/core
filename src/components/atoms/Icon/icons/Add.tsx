import React from 'react';
export const Add = {
  viewBox: '0 0 24 24',
  jsx: <path fill="currentColor" d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z" />,
};
