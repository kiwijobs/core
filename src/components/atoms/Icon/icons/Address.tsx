import React from 'react';

export const Address = {
  viewBox: '0 0 13 16',
  jsx: (
    <g fill="none" fillRule="evenodd">
      <path
        stroke="currentColor"
        d="M8.865 6c0 1.381-1.167 2.5-2.607 2.5C4.818 8.5 3.65 7.381 3.65 6s1.167-2.5 2.608-2.5c1.44 0 2.607 1.119 2.607 2.5z"
      />
      <path
        fill="currentColor"
        d="M6.78 15.5c0 .276-.234.5-.522.5a.511.511 0 0 1-.522-.5c0-.276.234-.5.522-.5.288 0 .521.224.521.5"
      />
      <path
        stroke="currentColor"
        d="M6.258 15.5s2.948-3.802 4.172-5.469C11.653 8.364 11.994 7.33 11.994 6c0-3.038-2.568-5.5-5.736-5.5C3.089.5.52 2.962.52 6c0 1.33.342 2.364 1.565 4.031C3.309 11.698 6.258 15.5 6.258 15.5"
      />
    </g>
  ),
};
