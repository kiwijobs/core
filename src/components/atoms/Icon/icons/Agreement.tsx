import React from 'react';

export const Agreement = {
  viewBox: '0 0 34 34',
  jsx: (
    <g fill="none" fillRule="evenodd">
      <circle cx={17} cy={17} r={17} fill="currentColor" />
      <g transform="rotate(-135 11.018 10.433)">
        <path
          fill="#283238"
          d="M0 5.746h3.946v10.936c0 .737-.597 1.335-1.334 1.335H1.335A1.335 1.335 0 010 16.682V5.746z"
        />
        <path fill="#283238" d="M0 5.746L1.973 0 3.946 5.746z" />
        <path fill="currentColor" d="M0 5.191H3.946V6.191H0z" />
        <path fill="currentColor" d="M0 14.071H3.946V15.071H0z" />
      </g>
    </g>
  ),
};
