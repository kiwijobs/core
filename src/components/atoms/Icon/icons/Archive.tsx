import React from 'react';

export const Archive = {
  viewBox: '0 0 24 24',
  jsx: (
    <path
      fill="currentColor"
      fillRule="evenodd"
      d="M6 7h12v2.25H6zM6.8 10v8h10.4v-8H6.8zm6.6 2.25h-2.8c-.663 0-1.2-.504-1.2-1.125 0-.207.18-.375.4-.375.22 0 .4.168.4.375s.18.375.4.375h2.8c.22 0 .4-.168.4-.375s.18-.375.4-.375c.22 0 .4.168.4.375 0 .621-.537 1.125-1.2 1.125z"
    />
  ),
};
