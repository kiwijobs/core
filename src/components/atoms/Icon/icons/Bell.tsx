import React from 'react';

export const Bell = {
  viewBox: '0 0 18 22',
  jsx: (
    <path
      fill="currentColor"
      stroke="currentColor"
      fillRule="nonzero"
      d="M8.981 20.986c-.603 0-1.119-.227-1.547-.681-.428-.454-.642-1-.642-1.64h4.378c0 .64-.214 1.186-.642 1.64-.428.454-.944.68-1.547.68zm7.683-3.84H1.278l.113-.184.263-.39.646-.937c.272-.403.567-.848.882-1.335.527-.811.79-1.7.79-2.668V6.967c0-.804.126-1.556.379-2.255.252-.7.603-1.31 1.052-1.83.449-.522.979-.935 1.59-1.24a4.34 4.34 0 0 1 1.967-.457c.716 0 1.379.152 1.99.457.61.305 1.14.718 1.588 1.24.45.52.804 1.13 1.063 1.83.26.7.39 1.45.39 2.255v4.665c0 .997.249 1.909.747 2.735.498.826.978 1.536 1.441 2.132l.253.323c.084.112.161.22.232.324z"
    />
  ),
};
