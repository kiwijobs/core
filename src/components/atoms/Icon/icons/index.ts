export * from './Add';
export * from './Address';
export * from './Agreement';
export * from './Archive';
export * from './Bell';
export * from './Check';
export * from './Chevron';
export * from './Clock';
export * from './Close';
export * from './Copy';
export * from './Disposition';
export * from './Document';
export * from './Duplicate';
export * from './GoodEmployer';
export * from './Hamburger';
export * from './Info';
export * from './Kiwi';
export * from './Link';
export * from './More';
export * from './Olx';
export * from './Pen';
export * from './Remove';
export * from './Salary';
export * from './Search';
export * from './Star';
export * from './Trash';
export * from './User';
