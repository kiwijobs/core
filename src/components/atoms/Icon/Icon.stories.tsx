import React from 'react';
import { storiesOf } from '@storybook/react';
import { Box, Flex } from '../../quarks';
import { Icon, IconProps } from './Icon';
import * as icons from './icons';

const StoryWrapper = (props: any) => (
  <Flex
    sx={{
      alignItems: 'center',
      flexWrap: 'wrap',
      m: 2,
    }}
    {...props}
  />
);
const SIcon = ({ text, ...props }: { text?: any } & IconProps) => (
  <Flex
    sx={{
      border: 1,
      borderRadius: 1,
      bg: 'white',
      color: '1',
      m: 1,
      p: 2,
      size: '8rem',
      alignItems: 'center',
      justifyContent: 'center',
      flexDirection: 'column',
    }}
  >
    <Icon {...props} />
    {text}
  </Flex>
);

storiesOf('Atoms|Icons', module)
  .addDecorator(story => <StoryWrapper>{story()}</StoryWrapper>)
  .add('Default', () => {
    const array = Object.keys(icons) as Array<keyof typeof icons>;

    return (
      <>
        {array.map(name => (
          <SIcon
            name={name}
            text={
              <Box mt={2} color="2" fontScale={1}>
                {name}
              </Box>
            }
          ></SIcon>
        ))}
      </>
    );
  })
  .add('Add', () => (
    <>
      <SIcon name="Add" />
      <SIcon
        sx={{
          border: 1,
          borderRadius: 1,
          borderColor: '1',
        }}
        name="Add"
      />
      <SIcon
        sx={{
          border: 1,
          borderRadius: '50%',
          borderColor: '1',
        }}
        name="Add"
      />
      <SIcon
        sx={{
          borderRadius: '50%',
          bg: 'primary',
          color: 'white',
        }}
        name="Add"
      />
    </>
  ))
  .add('Check', () => (
    <>
      <SIcon name="Check" />
      <SIcon
        sx={{
          border: 1,
          borderRadius: 1,
          borderColor: '1',
          p: 1,
        }}
        name="Check"
      />
      <SIcon
        sx={{
          borderRadius: 1,
          bg: 'primary',
          color: 'white',
          p: 1,
        }}
        name="Check"
      />
      <SIcon
        sx={{
          borderRadius: '50%',
          bg: 'primary',
          color: 'white',
          p: 1,
        }}
        name="Check"
      />
    </>
  ))
  .add('Close', () => (
    <>
      <SIcon p={1} name="Close" />
      <SIcon
        sx={{
          border: 1,
          borderRadius: 1,
          borderColor: '1',
          p: 1,
        }}
        name="Close"
      />
      <SIcon
        sx={{
          border: 1,
          borderRadius: '50%',
          borderColor: 'primary',
          color: 'primary',
          p: 1,
        }}
        name="Close"
        p={1}
      />
      <SIcon
        sx={{
          borderRadius: '50%',
          bg: 'primary',
          color: 'white',
          p: 1,
        }}
        name="Close"
      />
    </>
  ))
  .add('Info', () => (
    <>
      <SIcon name="Info" />
      <SIcon
        sx={{
          border: 1,
          borderRadius: '50%',
          borderColor: '1',
        }}
        name="Info"
      />
      <SIcon
        sx={{
          border: 1,
          borderRadius: '50%',
          bg: '1',
          color: 'white',
        }}
        name="Info"
      />
    </>
  ));
