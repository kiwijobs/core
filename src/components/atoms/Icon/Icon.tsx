import React, { forwardRef } from 'react';
import { Box, BoxProps } from '../../quarks';
import * as icons from './icons';

type Direction = 'UP' | 'RIGHT' | 'DOWN' | 'LEFT';

export type TIconName = keyof typeof icons;

export interface IconProps extends BoxProps<HTMLOrSVGElement> {
  name: TIconName;
  direction?: Direction;
}

const getDirection = (direction: Direction) => {
  switch (direction) {
    case 'UP':
      return;
    case 'RIGHT':
      return 'rotate(90deg)';
    case 'DOWN':
      return 'rotate(180deg)';
    case 'LEFT':
      return 'rotate(-90deg)';
    default:
      return;
  }
};

export const Icon = forwardRef<HTMLOrSVGElement, IconProps>(
  ({ name, direction = 'UP', sx, ...props }, ref) => {
    const { jsx, ...rest } = icons[name] || {};

    if (!jsx) {
      console.warn('No icon with a name: ' + name);
      return null;
    }

    return (
      <Box
        as="svg"
        ref={ref}
        sx={{
          size: '2.4rem',
          flex: '0 0 auto',
          transform: getDirection(direction),
          ...sx,
        }}
        {...rest}
        {...props}
      >
        {jsx}
      </Box>
    );
  }
);
