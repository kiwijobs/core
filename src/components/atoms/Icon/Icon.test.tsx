import React from 'react';
import { render } from '@testing-library/react';
import { wrapper } from '../../../__testUtils__';
import { Icon } from './Icon';
import * as icons from './icons';

describe('Icon', () => {
  it('Renders icon by name', () => {
    const array = Object.keys(icons) as Array<keyof typeof icons>;
    array.forEach(name => {
      expect(render(<svg>{icons[name].jsx}</svg>).container).toMatchSnapshot();
    });
  });

  it('Should render icon correctly', () => {
    const array = Object.keys(icons) as Array<keyof typeof icons>;
    array.forEach(name => {
      const { firstElementChild } = render(<Icon name={name} />).container;
      expect(firstElementChild!.tagName).toBe('svg');
      expect(firstElementChild).toHaveStyleRule('width', '2.4rem');
      expect(firstElementChild).toHaveStyleRule('height', '2.4rem');
      expect(firstElementChild).toHaveStyleRule('flex', '0 0 auto');
    });
  });

  it('Should render icon direction correctly', () => {
    const array = Object.keys(icons) as Array<keyof typeof icons>;
    array.forEach(name => {
      const { firstElementChild } = render(<Icon name={name} />).container;
      expect(firstElementChild!.tagName).toBe('svg');
      expect(firstElementChild).not.toHaveStyleRule('transform');
    });
    array.forEach(name => {
      const { firstElementChild } = render(<Icon name={name} direction="UP" />).container;
      expect(firstElementChild!.tagName).toBe('svg');
      expect(firstElementChild).not.toHaveStyleRule('transform');
    });
    array.forEach(name => {
      const { firstElementChild } = render(<Icon name={name} direction="RIGHT" />).container;
      expect(firstElementChild!.tagName).toBe('svg');
      expect(firstElementChild).toHaveStyleRule('transform', 'rotate(90deg)');
    });
    array.forEach(name => {
      const { firstElementChild } = render(<Icon name={name} direction="DOWN" />).container;
      expect(firstElementChild!.tagName).toBe('svg');
      expect(firstElementChild).toHaveStyleRule('transform', 'rotate(180deg)');
    });
    array.forEach(name => {
      const { firstElementChild } = render(<Icon name={name} direction="LEFT" />).container;
      expect(firstElementChild!.tagName).toBe('svg');
      expect(firstElementChild).toHaveStyleRule('transform', 'rotate(-90deg)');
    });
  });

  it('Warns if no icon', () => {
    console.warn = jest.fn();

    expect(
      render(<Icon name={'ThereShouldBeNoIconWithThatName' as any} />, { wrapper }).container
        .firstElementChild
    ).toBeNull();
    expect(console.warn).toBeCalledTimes(1);
  });
});
