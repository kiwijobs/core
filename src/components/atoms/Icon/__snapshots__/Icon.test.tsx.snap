// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Renders icon by name 1`] = `
<div>
  <svg>
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 2`] = `
<div>
  <svg>
    <path
      d="M8.981 20.986c-.603 0-1.119-.227-1.547-.681-.428-.454-.642-1-.642-1.64h4.378c0 .64-.214 1.186-.642 1.64-.428.454-.944.68-1.547.68zm7.683-3.84H1.278l.113-.184.263-.39.646-.937c.272-.403.567-.848.882-1.335.527-.811.79-1.7.79-2.668V6.967c0-.804.126-1.556.379-2.255.252-.7.603-1.31 1.052-1.83.449-.522.979-.935 1.59-1.24a4.34 4.34 0 0 1 1.967-.457c.716 0 1.379.152 1.99.457.61.305 1.14.718 1.588 1.24.45.52.804 1.13 1.063 1.83.26.7.39 1.45.39 2.255v4.665c0 .997.249 1.909.747 2.735.498.826.978 1.536 1.441 2.132l.253.323c.084.112.161.22.232.324z"
      fill="currentColor"
      fill-rule="nonzero"
      stroke="currentColor"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 3`] = `
<div>
  <svg>
    <path
      d="M1.707 6.293A1 1 0 1 0 .293 7.707l5 5a1 1 0 0 0 1.414 0l11-11A1 1 0 1 0 16.293.293L6 10.586 1.707 6.293z"
      fill="currentColor"
      fill-rule="nonzero"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 4`] = `
<div>
  <svg>
    <path
      d="M10.59 7.41L6 2.83 1.41 7.41 0 6l6-6 6 6z"
      fill="currentColor"
      fill-rule="nonzero"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 5`] = `
<div>
  <svg>
    <path
      d="M1.333 1.333l9.428 9.428M10.667 1.333l-9.334 9.334"
      fill="none"
      fill-rule="evenodd"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 6`] = `
<div>
  <svg>
    <path
      d="M0 0v12h9V3.2L5.727 0z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 7`] = `
<div>
  <svg>
    <path
      d="M10 9h2v8h-2zM10 5h2v2h-2z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 8`] = `
<div>
  <svg>
    <path
      d="M19 13H5v-2h14v2z"
      fill="currentColor"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 9`] = `
<div>
  <svg>
    <path
      d="M12 4.584l-4.22-.526L6 0 4.22 4.058 0 4.584l3.12 3.033L2.292 12 6 9.817 9.708 12 8.88 7.617z"
      fill="currentColor"
      fill-rule="nonzero"
    />
  </svg>
</div>
`;

exports[`Icon Renders icon by name 10`] = `
<div>
  <svg>
    <path
      d="M6 0a3 3 0 1 0 0 6 3 3 0 0 0 0-6zm0 7.5c-4.575 0-6 3-6 3V12h12v-1.5s-1.425-3-6-3z"
      fill="currentColor"
      fill-rule="nonzero"
    />
  </svg>
</div>
`;
